darts/timeseries.py

```diff
@@ -140,156 +582,251 @@ def pd_dataframe(self, copy=True) -> pd.DataFrame:
+            or an integer (if indexed by RangeIndex)
+        """
+        return self._time_index[-1]
 
     def first_value(self) -> float:
         """
         Returns
         -------
         float
-            The first value of this univariate time series
+            The first value of this univariate deterministic time series
         """
         self._assert_univariate()
-        return self._values[0][0]
+        self._assert_deterministic()
+        return float(self._xa[0, 0, 0])
 
     def last_value(self) -> float:
         """
         Returns
         -------
         float
-            The last value of this univariate time series
+            The last value of this univariate deterministic time series
         """
         self._assert_univariate()
-        return self._values[-1][0]
+        self._assert_deterministic()
+        return float(self._xa[-1, 0, 0])
 
     def first_values(self) -> np.ndarray:
         """
         Returns
         -------
         np.ndarray
-            The first values of every component of this time series
+            The first values of every component of this deterministic time series
         """
-        return self._values[0]
+        self._assert_deterministic()
+        return self._xa.values[0, :, 0].copy()
 
     def last_values(self) -> np.ndarray:
         """
         Returns
         -------
         np.ndarray
-            The last values of every component of this time series
+            The last values of every component of this deterministic time series
         """
-        return self._values[-1]
+        self._assert_deterministic()
+        return self._xa.values[-1, :, 0].copy()
 
-    def values(self, copy=True) -> np.ndarray:
+    def values(self, copy=True, sample=0) -> np.ndarray:
         """
+        Returns a 2-D Numpy array of dimension (time, component), containing this series' values for one sample.
+        If this series is deterministic, it contains only one sample and only `sample=0` can be used.
+
         Parameters
         ----------
         copy
-            Whether to return a copy of the values. Leave it to True unless you know what you are doing.
+            Whether to return a copy of the values, otherwise returns a view.
+            Leave it to True unless you know what you are doing.
 
         Returns
         -------
         numpy.ndarray
             The values composing the time series.
         """
+        raise_if(self.is_deterministic and sample != 0, 'This series contains one sample only (deterministic),'
+                                                        'so only sample=0 is accepted.', logger)
         if copy:
-            return np.copy(self._values)
+            return np.copy(self._xa.values[:, :, sample])
         else:
-            return self._values
+            return self._xa.values[:, :, sample]
 
-    def univariate_values(self, copy=True) -> np.ndarray:
+    def all_values(self, copy=True) -> np.ndarray:
```

Comment:
If we change it anyway we should call it simply values and make it a property method as well. Would be closer to pandas and also xarray implementation.

Reply:
But this "copy" argument is important...